File: 1.VSC/3.JAVA Workspace/Stack/Stack.java
package Stack;

public class Stack{
    int[] arr =new int[100];
    int topIndex = 0;
    
    public boolean empty(){
        return this.topIndex == 0;
    }

    public void push(int x){
        this.topIndex++;
        this.arr[topIndex] = x;
    }

    public int top(){
        return this.arr[this.topIndex];
    }

    public void pop(){
        this.topIndex--;
    }
}

File: 1.VSC/1.C Workspace/C-Stack/Stack.c
#include<stdlib.h>

struct nodeee{
    int val;
    struct nodeee* next;
};

typedef struct nodeee* Stack;

void makenullStack(Stack *a){
    (*a) = (Stack)malloc(sizeof(Stack));
    (*a)->next = NULL;
}

int emptyStack(Stack *a){
    return (*a)->next == NULL;
}

void push(Stack *a, int x){
    Stack pos = *a;
    while (pos->next != NULL)
    {
        pos = pos->next;
    }
    Stack temp = (Stack)malloc(sizeof(Stack));
    temp->val = x;
    temp->next = pos->next;
    pos->next = temp;
}

int top(Stack *a){
    Stack pos = *a;
    while (pos->next != NULL)
    {
        pos = pos->next;
    }
    return pos->val;
}

void pop(Stack *a){
    Stack pos = *a;
    while (pos->next->next != NULL)
    {
        pos = pos->next;
    }
    
    free(pos->next);
    pos->next = NULL;
}




